Support hourly index patterns

jmacdone opened the pull request:
## Description

Collect otherwise missing data for top_count_keys for those indexing hourly with a %H pattern, eg. `logstash-%Y.%m.%d.%H`
Previously format_index() assumed daily was the most granular indexing.

See also https://github.com/jertel/elastalert2/discussions/1326

## Checklist

<!--
The following checklist items must be completed before PRs can be merged. 
-->

- [x] I have reviewed the [contributing guidelines](https://github.com/jertel/elastalert2/blob/master/CONTRIBUTING.md).
- [x] I have included unit tests for my changes or additions.
- [x] I have successfully run `make test-docker` with my changes.
- [x] I have manually tested all relevant modes of the change in this PR.
- [ ] I have updated the [documentation](https://elastalert2.readthedocs.io).
- [x] I have updated the [changelog](https://github.com/jertel/elastalert2/blob/master/CHANGELOG.md).


## Questions or Comments

test-docker
>   py311: OK (575.05=setup[483.55]+cmd[89.54,1.97] seconds)
>  docs: OK (481.68=setup[466.33]+cmd[15.35] seconds)
>  congratulations :) (1056.81 seconds)

jertel reviewed (changes requested):
Thanks for adding the test coverage and cleaning up the logic in the original function. I included a suggested logic change in my feedback.

jmacdone commented:
- committed @jertel 's suggested changes with fix-ups
 - updated the unit test to reflect intended meaning of the `add_extra` param
 - rebased from master
